### ISSUE TITLE
Unrelated retrieval data in the training set

### ISSUE THREAD

**Issue by ryoungj** [OpenForecaster/scaling-forecasting-training#5]
Hi,

Congrats on the great work! 

I was looking at the released dataset and have a question: it seems that most retrieved articles are unrelated with the question in the training set (but not the val/test set). I asked Claude to make a simple relevance eval and here is the stat:

| Split | Avg relevance (0-10) | Notes |
|-------|---------------------|-------|
| train | **0.75** | Nearly all articles are completely unrelated to the question (random Forbes/CNBC articles from 2022-2023). Only 2/20 samples had any relevance. |
| validation | **6.15** | Most articles are topically relevant. ~10/20 scored 7+. Articles tend to be from 2025 news sources matching the question topics. |
| test | **5.65** | Similar to validation. ~8/20 scored 7+. Mix of directly relevant and loosely related articles. |

Is this expected or is it an issue in the uploaded train set?

Thanks!

**Comment by nikhilchandak** [comment]
Hi Yangjun, thanks for bringing up this issue! This is not expected (all splits should have similar relevance ideally). I will look into this and get back to you soon.

**Comment by shash42** [comment]
Hi Yangjun, our train set had retrieval over a subset of the news corpus (as mentioned in the paper) just because it was a larger number of retrievals to make. Thanks for pointing out this leads to significant degradation in quality. We will soon release an updated version of the training set with retrieval over the full corpus, and also quantify the difference in quality with forecasting evals/training

**Comment by shash42** [comment]
Thanks for pointing out this issue. https://huggingface.co/datasets/nikhilchandak/OpenForesight now has the updated train set with retrieval over the full document corpus, and the chunks are much more relevant to the question.

**Comment by nikhilchandak** [comment]
Hi @ryoungj,

Thank you so much for noticing and bringing this up! We had initially done retrieval over a very small set of news sources (< 10) to iterate faster but forgot to update the train set later with full news corpus.  We provide some stats below.

Evaluating Qwen3-8B on 1000 random questions from the train set, we observe the new retrieval improves accuracy 3x thus showing the new retrieval has much more relevant chunks and also the model says "Unknown" answer much less. 

| Metric | OLD retrieval | New retrieval |
|--------|--------------|---------------|
| Accuracy | 10.2% | **30.1%** |
| Unknown % | 46.1% | **15.0%** |

We also performed a RL training run of Qwen3-8B on this updated train set and it also improves performance on the test set (below) by 2% on accuracy and 0.02 brier points! 

<img width="3432" height="3288" alt="Image" src="https://github.com/user-attachments/assets/1b1ec453-9dc8-4c3b-8299-4ec80532d4ce" />

We have now updated the train set on Huggingface with full retrieval.

Cheers!
